plutus-core/plutus-ir/src/PlutusIR/Compiler/Types.hs

```diff
@@ -133,7 +137,7 @@ data CompilationCtx uni fun a = CompilationCtx {
 makeLenses ''CompilationCtx
 
 toDefaultCompilationCtx
-    :: (Default (PLC.BuiltinSemanticsVariant fun), Default (PLC.CostingPart uni fun))
+    :: (Ord fun, Default (PLC.BuiltinSemanticsVariant fun), Default (PLC.CostingPart uni fun))
```

Comment:
This `Default CostingPart` always makes me uneasy. It sucks that we have to pass some fake (zero) consting just so we can have a runtime available for the EvaluateBuiltins pass (and perhaps other passes as well)

Reply:
Is there another way to do it? Costing _is_ a part of the evaluation semantics of a builtin.

Reply:
I guess in principle we could make it optional? But then we'd need to pass `Maybe cost` around in various places which would have overhead in the machine, so this seems better.

Reply:
> I guess in principle we could make it optional? 

![image](https://github.com/input-output-hk/plutus/assets/10480926/2c74d91f-79c0-4e2b-8afc-ecd2577d2316)

Reply:
@bezirg 

> But the compiler does not make use of the costing part at all. It's just for compile-time evaluation, disregarding the costs (more precisely treating all as zero-cost)

Good point, though. I'll think about it, thank you.

Reply:
If you only need compile-time evaluation, you can use the first two fields of `BuiltinMeaning` like we do in the well-typed Hedgehog program [generators](https://github.com/input-output-hk/plutus/blob/de48aa5ddce4dc3c571e84fb763e9e88cb7239c2/plutus-core/testlib/PlutusCore/Generators/Hedgehog/Denotation.hs#L39-L42) for example. Although it's probably not a good idea, since you probably don't want to roll out your own lifting and unlifting logic etc. Maybe we should indeed have some middle-ground. 

Although I still feel like insisting everywhere that costing is a part of operational semantics of builtins will save us so much trouble in the long run that it may not be optimal to break this rule even when it makes sense. I may be wrong about that in this particular case, though.